lua/el/extensions.lua

```diff
@@ -75,4 +75,170 @@ extensions.mode = function(_, _)
   )
 end
 
+-- adding it as a function just for better clarity
+-- on whats going on
+local file_extension = function(filename)
+  return filename:match('%.(%w+)$')
+end
+
+extensions.file_icon = function(_, bufnr)
```

Comment:
also, it should be `buffer` not `bufnr`.

`buffer` is an "object" like item that contains some metadata about the buffer so that you don't have to look it up every time.

Reply:
Oh, wasn't aware of that. Changing it now

Reply:
yeah, i was already using the object, but changing the param name to the actual object name makes more sense

Reply:
right, that's what I was meaning to say. `bufnr` has a special meaning (the buffer number), so I try and be very careful where I use it.